index.bs

```diff
@@ -755,8 +760,10 @@ spec.  It would be nice to unify these in the future.</p>
 To <dfn>clear cookies for host</dfn> given a [=host=] |host|, perform the
 following steps:
 
-1. Let |cookieList| be the set of cookies from the [=cookie store=] whose
-    domain attribute is a [=domain-match=] with |host|.
+1. Let |cookieList| be a set of cookies, initially empty.
+1. [=list/For each=] cookie |cookie| in the [=cookie store=], if either of the following is true, add |cookie| to |cookieList|:
+     - |cookie| is not [=partitioned cookie|partitioned=], and |cookie|'s [=domain attribute=] is a [=domain-match=] with |host|; or
+     - |cookie| is [=partitioned cookie|partitioned=], and |cookie|'s [=partition key=] is equal to |host|'s [=obtain a site|site=].
```

Comment:
This doesn't work. "obtain a site" takes an origin, not a host.

And yeah, doing this would require a new hook in https://github.com/johannhof/draft-annevk-johannhof-httpbis-cookies/ and also some decision around whether to dispatch events I guess? Probably best to start thinking about that new and leave some XXX/TODO marker here and file an issue.

Reply:
> Yeah, I think this might need to adapt to ACB.

What is ACB?

> This doesn't work. "obtain a site" takes an origin, not a host.

Thanks for the catch — I've updated this line.

> And yeah, doing this would require a new hook in https://github.com/johannhof/draft-annevk-johannhof-httpbis-cookies/ and also some decision around whether to dispatch events I guess? Probably best to start thinking about that n[o]w and leave some XXX/TODO marker here and file an issue.

Happy to add a note about the intersection with the linked draft, but it's not clear to me what exactly I should be noting.

Reply:
This still does not seem correct to me. Or is host always canonicalized to a registrable domain beforehand or some such?

Reply:
All the hosts that this algorithm is called for are from the user activation map, which indeed only stores site hosts, but that connection is not obvious as written. I changed the algorithm name to "clear cookies for site host" for clarity.

Reply:
Hmm, I think it would be better to undo that commit (site host isn't really a thing) and add an assert that host's registrable domain is null.

Reply:
> (site host isn't really a thing)

I see.

> add an assert that host's registrable domain is null.

Based on my reading of https://url.spec.whatwg.org/#example-host-psl, I think I should assert that `host`'s registrable domain is still equal to `host` (e.g., in the table, `example.com`'s registrable domain is `example.com`). But totally open to the possibility that I've misunderstood something.

Reply:
You're right, that's not going to work. But asserting it's itself is also not going to work as per the `gitHub.io` example. I guess we can assert that it's either equal to itself or null. That's still useful and would fail for inputs like `www.example.com`, which is mainly what we want to catch.

Reply:
Good point, and thanks for going back and forth with me. Updated in latest push.

Reply:
@annevk Have your concerns been addressed?

Reply:
I think so, but as Johann pointed out the ancestor-chain-bit (ACB) needs addressing and Ben is correct about storage in general.

Reply:
Can we rephrase this to just look at the top-level site contained in the partition key?  That way ACB and any future additions to the partition key would still be handled correctly.

Reply:
I think that works.

And I think as long as this works similarly to the end user deleting an entire site there's no problem either. I got a bit worried about giving a top-level site the ability to delete cookies outside of its direct control, but I think it holds up.

Reply:
> Can we rephrase this to just look at the top-level site contained in the partition key?

Added in latest push. Working on the local storage piece.

Reply:
Updated the deletion of non-cookie storage to account for the addition of top-level site to storage keys in whatwg/storage#144.